runner/src/provisioner/provisioner.ts

```diff
@@ -44,6 +45,23 @@ export default class Provisioner {
       .replace(/\//g, '0');
   }
 
+  isUserApiProvisioned (accountId: string, functionName: string): boolean {
+    const accountIndexers = this.#hasBeenProvisioned[accountId];
+    if (!accountIndexers) { return false; }
+    return accountIndexers[functionName];
+  }
+
+  private setProvisioned (accountId: string, functionName: string): void {
+    const accountIndexers = this.#hasBeenProvisioned[accountId] ?? {};
+    this.#hasBeenProvisioned = {
+      ...this.#hasBeenProvisioned,
+      [accountId]: {
+        ...accountIndexers,
+        [functionName]: true
+      }
+    };
```

Comment:
```suggestion
    this.#hasBeenProvisioned[accountId] ??= {};
    this.#hasBeenProvisioned[accountId][functionName] = true;
```
We can use Nullish coalescing assignment (`??=`) to ensure the account record exists, and just mutate the single field rather than replacing the entire object.